src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+} from 'react';
+import cn from 'classnames';
+import { TodoContext } from '../../TodoContext';
+import { Todo } from '../../types/Todo';
+
+export const TodoItem: React.FC<{ todo: Todo }> = React.memo(({ todo }) => {
+  const {
+    isLoading,
+    editingTodo,
+    handleEditing,
+    setHandleEditing,
+    setIsChosenToRename,
+    setEditingTodo,
+    handleDelete,
+    makeTodoCompleted,
+  } = useContext(TodoContext);
+
+  return (
+    <div
+      data-cy="Todo"
+      className={cn('todo', {
+        completed: todo.completed,
+      })}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={() => null}
+          onClick={() => {
+            makeTodoCompleted(todo.id, todo.completed);
+          }}
+        />
+      </label>
+
+      <span
+        data-cy="TodoTitle"
+        className="todo__title"
+        role="button"
+        tabIndex={0}
+        onDoubleClick={() => {
+          if (handleEditing === 0) {
+            setIsChosenToRename(todo.id);
+            setEditingTodo(todo.title);
+            setHandleEditing(todo.id);
+          }
```

Comment:
it looks like `setIsChosenToRename(todo.id);` and ` setHandleEditing(todo.id);` are the same.
and have the same functionality, you can remove one, and the other one use instead of it